client/my-sites/stats/stats-chart-tabs/index.jsx

```diff
@@ -26,51 +26,48 @@ import {
 	getSiteStatsNormalizedData,
 } from 'state/stats/lists/selectors';
 import { recordGoogleEvent } from 'state/analytics/actions';
-import { rangeOfPeriod } from 'state/stats/lists/utils';
 import { getSiteOption } from 'state/sites/selectors';
+import { formatDate, getQueryDate } from './utility';
 
 class StatModuleChartTabs extends Component {
-	constructor( props ) {
-		super( props );
-		const activeTab = this.getActiveTab();
-		const activeCharts = activeTab.legendOptions ? activeTab.legendOptions.slice() : [];
-		this.state = {
-			activeLegendCharts: activeCharts,
-			activeTab: activeTab,
-		};
-	}
+	static propTypes = {
+		data: PropTypes.arrayOf(
+			PropTypes.shape( {
+				comments: PropTypes.number,
+				labelDay: PropTypes.string,
+				likes: PropTypes.number,
+				period: PropTypes.string,
+				posts: PropTypes.number,
+				visitors: PropTypes.number,
+				visits: PropTypes.number,
+			} )
+		),
+		isActiveTabLoading: PropTypes.bool,
+	};
 
-	componentWillReceiveProps( nextProps ) {
-		const activeTab = this.getActiveTab( nextProps );
-		const activeCharts = activeTab.legendOptions ? activeTab.legendOptions.slice() : [];
-		if ( activeTab !== this.state.activeTab ) {
-			this.setState( {
-				activeLegendCharts: activeCharts,
-				activeTab: activeTab,
-			} );
+	state = {
+		activeLegendCharts: null,
+		activeTab: null,
+	};
+
+	static getDerivedStateFromProps( props, state ) {
+		const activeTab = StatModuleChartTabs.getActiveTab( props );
+		const activeLegendCharts = activeTab.legendOptions ? activeTab.legendOptions.slice() : [];
```

Comment:
no need to perform this calculation outside of the below `if` statement - if we move it inside then it will only allocate and free the memory when it needs to vs. on every render

Reply:
Whoops, overlooked this one.